### PR TITLE
Giropay Integration

### PR DESCRIPTION
**Pull request by jezao**

### CODE REVIEW

File: examples/inc/giropayconfig.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+// # Configuration
+
+// The payment SDK needs some basic configuration regarding connectivity and merchant account IDs.
+
+// ## Required objects
+
+// To include the necessary files, we use the composer for PSR-4 autoloading.
+require __DIR__ . '/../vendor/autoload.php';
```
Comment (jakubpolomsky) [inline examples/inc/giropayconfig.php]: this path would yield a - file not found - error, suggestion `/../../vendor/autoload.php`

File: examples/GiroPay/pay.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+$bankAccount = new BankAccount();
+$bankAccount->setBic("GENODETT488");
+
+
+//// The redirect URLs determine where the consumer should be redirected by GiroPay after approval/cancellation.
+$redirectUrls = new Redirect(getUrl('return.php?status=success'), getUrl('return.php?status=cancel'));
+//
+////// As soon as the transaction status changes, a server-to-server notification will get delivered to this URL.
+//$notificationUrl = getUrl('notify.php');
```
Comment (jakubpolomsky) [inline examples/GiroPay/pay.php]: please uncomment the `$notificationUrl` and remove unnecessary slashes (`/`)

File: examples/GiroPay/pay.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+// # Giropay payment transaction
+// This example displays the usage payments for payment method Giropay.
+
+// ## Required objects
+
+// To include the necessary files, we use the composer for PSR-4 autoloading.
+
+require __DIR__ . '/../../vendor/autoload.php';
+require __DIR__ . '/../inc/common.php';
+require __DIR__ . '/../inc/config.php';
```
Comment (jakubpolomsky) [inline examples/GiroPay/pay.php]: giropayconfig should be required, not the generic config

File: examples/GiroPay/notify.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<?php
+// # Giropay notification
+
+// Wirecard sends a server-to-server request regarding any changes in the transaction status.
+
+// ## Required objects
+
+// To include the necessary files, we use the composer for PSR-4 autoloading.
+require __DIR__ . '/../../vendor/autoload.php';
+require __DIR__ . '/../inc/config.php';
```
Comment (jakubpolomsky) [inline examples/GiroPay/notify.php]: giropayconfig should be required, not the generic one

File: examples/GiroPay/return.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+// # GiroPay return after transaction
+
+// The consumer gets redirected to this page after a GiroPay transaction.
+
+// ## Required objects
+
+// To include the necessary files, we use the composer for PSR-4 autoloading.
+require __DIR__ . '/../../vendor/autoload.php';
+require __DIR__ . '/../inc/common.php';
+require __DIR__ . '/../inc/config.php';
```
Comment (jakubpolomsky) [inline examples/GiroPay/return.php]: giropayconfig should be required, not the generic one

File: test/Transaction/GiroPayTransactionUTest.php [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+
+    public function testShouldReturnEndPointNotParent()
+    {
+        $returnType = $this->callMethod($this->tx, 'getEndpoint');
+
+        $this->assertEquals($returnType, '/engine/rest/paymentmethods/');
+    }
+
+
+//    public function reserveDataProvider()
```
Comment (jakubpolomsky) [inline test/Transaction/GiroPayTransactionUTest.php]: please remove commented out sections as there is no need for them to stay in the code

File: src/Transaction/Transaction.php [hunk 7]
```diff
@@ -482,6 +484,9 @@ protected function retrieveTransactionType()
             case Operation::CREDIT:
                 $transactionType = $this->retrieveTransactionTypeForCredit();
                 break;
+            case Operation::GETURL:
```
Comment (jakubpolomsky) [inline src/Transaction/Transaction.php]: remove this case please, get url has no back-end operations

File: src/Transaction/Transaction.php [hunk 8]
```diff
@@ -72,6 +72,8 @@
     const TYPE_VOID_PURCHASE = 'void-purchase';
     const TYPE_VOID_DEBIT= 'void-debit';
     const TYPE_DEPOSIT = 'deposit';
+    const TYPE_GET_URL = 'get-url';
```
Comment (jakubpolomsky) [inline src/Transaction/Transaction.php]: we don't need this case, please remove it

Comment (jezao) [reply src/Transaction/Transaction.php]: Jakub, What is the correct type? When i point to TYPE_DEBIT, i got this error:
THE SIGNATURE COULD NOT BE VERYFIED, CHECK LOG FOR DETAILS

Comment (jakubpolomsky) [reply src/Transaction/Transaction.php]: The type should be TYPE_DEBIT as you found out yourself. Today the giropay sandbox has some issues. That will be ok. Please continue with TYPE_DEBIT and let's hope the sandbox will start working.

**Comment by jakubpolomsky** [comment]
Thank you for your contribution! After a few minor changes and additions which couldn't have been clear for you I was happy to merge the pull request!

**Pull request merged**